src/core/mac/mac.cpp

```diff
@@ -782,25 +782,41 @@ void Mac::HandleBeginTransmit(void)
 
     if (error != kThreadError_None)
     {
-        TransmitDoneTask(false, kThreadError_Abort);
+        TransmitDoneTask(mTxFrame, false, kThreadError_Abort);
     }
 }
 
 extern "C" void otPlatRadioTransmitDone(otInstance *aInstance, RadioPacket *aPacket, bool aRxPending,
                                         ThreadError aError)
 {
     otLogFuncEntryMsg("%!otError!, aRxPending=%u", aError, aRxPending ? 1 : 0);
-    (void)aPacket;
-    aInstance->mThreadNetif.GetMac().TransmitDoneTask(aRxPending, aError);
+
+    aInstance->mThreadNetif.GetMac().TransmitDoneTask(aPacket, aRxPending, aError);
     otLogFuncExit();
 }
 
-void Mac::TransmitDoneTask(bool aRxPending, ThreadError aError)
+void Mac::TransmitDoneTask(RadioPacket *aPacket, bool aRxPending, ThreadError aError)
 {
     mMacTimer.Stop();
 
     mCounters.mTxTotal++;
 
+    Frame *packet = static_cast<Frame *>(aPacket);
+    Address addr;
+    packet->GetDstAddr(addr);
+
+    if (addr.mShortAddress == kShortAddrBroadcast)
+    {
+        // Broadcast packet
+        mCounters.mTxBroadcast++;
+    }
+    else
+    {
+        // Unicast packet
+        mCounters.mTxUnicast++;
```

Comment:
I'm not completely sure what you are asking.  I decided to count the unicast and broadcast packets here since this is also where the total Tx count is done (Line 802).